README.md

```diff
@@ -53,6 +53,19 @@ $ cd `guile -c "(display (%global-site-dir))"`
 $ wget https://raw.githubusercontent.com/jerry40/guile-simple-zmq/master/src/simple-zmq.scm
 ```
 
+Last, [Guile-Gcrypt](https://notabug.org/cwebber/guile-gcrypt) is needed
+to compute hash-based message authentication codes (HMACs) while
+communicating with Jupyter:
+```
+$ wget https://notabug.org/cwebber/guile-gcrypt/archive/v0.1.0.tar.gz
+$ tar xvf v0.1.0.tar.gz
+$ cd guile-gcrypt-0.1.0
```

Comment:
I get `guile-gcrypt` folder after unpacking 
I installed (Ubuntu 18.04) guile-2.2-dev and texinfo and executed
`./configure --prefix=/usr/ --with-libgcrypt-libdir=/usr/lib/x86_64-linux-gnu/` and it worked, but make returns an error:
```
cd "$am__cwd"; \
if /bin/bash /home/jerry/Downloads/guile-gcrypt/build-aux/missing makeinfo   -I . \
 -o guile-gcrypt.info guile-gcrypt.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd .; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd . && \
  $restore $backupdir/* `echo "./guile-gcrypt.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
./pre-inst-env  compile -Wunbound-variable -Warity-mismatch -Wformat -o "gcrypt/base16.go" "gcrypt/base16.scm"
./pre-inst-env: 33: exec: compile: not found
Makefile:1410: recipe for target 'gcrypt/base16.go' failed
```

Reply:
autoreconf:
```
jerry@MS-7721:~/Downloads/guile-gcrypt$ autoreconf -vfi
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
autoreconf: Leaving directory `.'
```
guild:
```
jerry@MS-7721:~/Downloads/guile-gcrypt$ guild --version
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /usr/bin/guild
;;; compiled /home/jerry/.cache/guile/ccache/2.2-LE-8-3.A/usr/bin/guild.go
guild (GNU Guile) 2.2.3
Copyright (C) 2017 Free Software Foundation, Inc.
```
hm...
```
jerry@MS-7721:~/Downloads/guile-gcrypt$ grep GUILD config.log
GUILD=''
```

Reply:
Looks like it is looking for `guld-2.2` command
```
jerry@MS-7721:~/Downloads/guile-gcrypt$ grep -i guild config.log
configure:2645: checking for guild-2.2
GUILD=''
```
Interesting

Reply:
Super weird.  Did you build Guile 2.2 by hand or are you using a distro-provided package?  If the latter, could it be that the distro developers modified either the `guile.m4` file (which contains Autoconf macros such as `GUILE_PKG`)?

In the meantime I suppose you have to go with the workaround I mentioned before.

Reply:
Now I get this message (guile can't find hmac module in the current dir?)
```
;;; compiling /home/jerry/.local/share/jupyter/kernels/guile/guile-jupyter-kernel.scm
;;; WARNING: compilation of /home/jerry/.local/share/jupyter/kernels/guile/guile-jupyter-kernel.scm failed:
;;; no code for module (hmac)
Backtrace:
          11 (primitive-load "/home/jerry/.local/share/jupyter/kerne…")
In ice-9/eval.scm:
   721:20 10 (primitive-eval (use-modules (simple-zmq) (json) (#) # …))
In ice-9/psyntax.scm:
  1235:36  9 (expand-top-sequence ((use-modules (simple-zmq) (#) …)) …)
  1182:24  8 (parse _ (("placeholder" placeholder)) ((top) #(# # …)) …)
   285:10  7 (parse _ (("placeholder" placeholder)) (()) _ c&e (eval) …)
In ice-9/boot-9.scm:
  3365:20  6 (process-use-modules _)
   222:29  5 (map1 (((simple-zmq)) ((json)) ((hmac)) ((srfi #)) (#)))
   222:29  4 (map1 (((json)) ((hmac)) ((srfi srfi-1)) ((srfi #))))
   222:17  3 (map1 (((hmac)) ((srfi srfi-1)) ((srfi srfi-13))))
  3366:31  2 (_ ((hmac)))
   2791:6  1 (resolve-interface _ #:select _ #:hide _ #:prefix _ # _ …)
In unknown file:
           0 (scm-error misc-error #f "~A ~S" ("no code for modu…" …) …)

ERROR: In procedure scm-error:
no code for module (hmac)
```

Reply:
This suggests that `hmac.scm` is not in Guile's _module search path_.

To fix that, you need to either set `GUILE_LOAD_PATH` to contain the current directory (it's not there by default, for security reasons), or to pass the `-L` flag to `guile`.  See [the manual](https://www.gnu.org/software/guile/manual/html_node/Load-Paths.html) for details.

HTH!

Reply:
Hi!
Yes I think I can avoid this, but I'd like to provide an installation process to users as easy as it can be. I'm afraid that people can be disappointed if the process is too difficult. Can we do anything in order to make it smoother?

Reply:
Hello!

I understand your concern.  I feel like `./configure && make && make install` is as "simple as it can be" in terms of build systems and "least surprise", but I agree that even that can be daunting to some users.

To me, given my own biases ;-), the best answer to such questions is packaging: guile-gcrypt is already in openSuSE, in AUR (unofficial Arch packages), and in Guix.  If more distros ship it, then it'll be easier for users building guile-kernel from source.

In the meantime, I don't have a better answer!

Ludo'.